chore(ai): improve `AGENTS.md` with verified prompts

hyf0 opened the pull request:
Refs:

- https://github.com/v8/v8/blob/f7c2751ff9e362b668f6c0fbbcf2df2fe68f1635/GEMINI.md
- https://www.anthropic.com/engineering/claude-code-best-practices

Prompts I didn't add

```
## Key Technologies
- **Rust**: Core bundling logic for performance
- **oxc**: Parser/analyzer/minifier
- **napi-rs**: Rust↔Node.js bridge
```

Becuase AI doesn't have prior knowledge about them(maybe they have?), they are only meaningful if you could give AI references/links to look up the code.  


```
### Testing Strategy

- **Rust**: Standard cargo test with cargo-insta for snapshots
- **Node.js**: Vitest-based test suites with fixture-driven testing
- **Compatibility**: Extensive rollup test suite for API compatibility
- **HMR testing**: Specialized dev server tests (`just test-node-hmr`)
- **Update snapshots**: Use `just test-update` or individual test commands with `--update`

### Common Gotchas

- Always use `just` commands instead of npm scripts for builds/tests
- The project requires both Rust and Node.js toolchains to be properly set up
- Some tests require git submodules to be initialized (`just setup-submodule`)
- Release builds are significantly slower but required for benchmarking
- Use `just debug-test` for faster Rust test iteration during development
- The `roll` command is comprehensive but slow - use specific commands during development

```

They are either repetition included in the PR or just nonsense.

```

# Stack-Based Development Workflow with Graphite

...
```

This one is too opinionated. Also I think subagent is better for such task. I keep encountering AI forget about these requirements due to the context is not enough.

```

## Verification Strategy (MANDATORY)

...

```

This one is another gem, so I'm still in the process of practicing it.

hyf0 commented:
* **#6533** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6533?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6533?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite: merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/rolldown?ref=merge-queue-instructions-in-stack-comment&prId=6533).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

hyf0 commented on AGENTS.md:
```diff
@@ -1,7 +1,67 @@
+       ↓
+NAPI Bindings (crates/rolldown_binding)
+       ↓
+Rust Core (crates/rolldown)
```
This's useful when you using prompt like

- Add an option xxx from rust core to ts api.
- Add an option xxx in rust core only.
	- Alt: Add an option xxx, don't expose it in ts api. (might risk adding code on binding layer
- Expose option field xxx from ts api.

Basically having the `Three-Layer Architecture` gives you the ability to refer to the layer you want AI to work on.

hyf0 commented on AGENTS.md:
```diff
@@ -1,7 +1,67 @@
+# Common Pitfalls & Best Practices
+
+- **Check surrounding code for conventions:** Before adding new code, always study the existing patterns, naming conventions, and architectural choices in the file and directory you are working in.
+- **Break large changes into tracked steps:** Decompose substantial work into manageable subtasks. Track progress to prevent scope creep and missed items. Use `TodoWrite` or similar tools to maintain a checklist.
```
- Check surrounding code for conventions
- Break large changes into tracked steps

are very effective prompts. 

`Check surrounding code for conventions` makes AI write code that follows style and even learns to write domain-specific code like napi-rs related code. In other words, it makes AI make a good plan before editing your code.

`Break large changes into tracked steps` makes AI write code following the plan. Can't say how much this is precious.

hyf0 commented on AGENTS.md:
```diff
@@ -1,7 +1,69 @@
+
+# Bash commands
+
+IMPORTANT: The project uses `just` as a task runner. Always prefer `just` commands over direct `npm` or `cargo` commands. Run `just --list` before using `npm` or `cargo` directly.
```
This is what you get from `Use release rolldown`

<img width="1158" height="336" alt="Image" src="https://github.com/user-attachments/assets/9e979041-ffb3-4dda-92aa-c051867be3c1" />

instead of `cargo build --release ...`

shulaoda commented:
### Merge activity

* **Oct 15, 6:28 AM UTC**: The merge label 'graphite: merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) once it meets the requirements.
* **Oct 15, 6:28 AM UTC**: `shulaoda` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).
* **Oct 15, 6:31 AM UTC**: Merged by the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).

hyf0 committed:
chore(ai): improve `AGENT.md` with verified prompts (#6533)

Refs:

- https://github.com/v8/v8/blob/f7c2751ff9e362b668f6c0fbbcf2df2fe68f1635/GEMINI.md
- https://www.anthropic.com/engineering/claude-code-best-practices

Prompts I didn't add

```
## Key Technologies
- **Rust**: Core bundling logic for performance
- **oxc**: Parser/analyzer/minifier
- **napi-rs**: Rust↔Node.js bridge
```

Becuase AI doesn't have prior knowledge about them(maybe they have?), they are only meaningful if you could give AI references/links to look up the code.

```
### Testing Strategy

- **Rust**: Standard cargo test with cargo-insta for snapshots
- **Node.js**: Vitest-based test suites with fixture-driven testing
- **Compatibility**: Extensive rollup test suite for API compatibility
- **HMR testing**: Specialized dev server tests (`just test-node-hmr`)
- **Update snapshots**: Use `just test-update` or individual test commands with `--update`

### Common Gotchas

- Always use `just` commands instead of npm scripts for builds/tests
- The project requires both Rust and Node.js toolchains to be properly set up
- Some tests require git submodules to be initialized (`just setup-submodule`)
- Release builds are significantly slower but required for benchmarking
- Use `just debug-test` for faster Rust test iteration during development
- The `roll` command is comprehensive but slow - use specific commands during development

```

They are either repetition included in the PR or just nonsense.

```

# Stack-Based Development Workflow with Graphite

...
```

This one is too opinionated. Also I think subagent is better for such task. I keep encountering AI forget about these requirements due to the context is not enough.

```

## Verification Strategy (MANDATORY)

...

```

This one is another gem, so I'm still in the process of practicing it.
